src/vos/tests/vts_io.c

```diff
@@ -659,27 +674,34 @@ io_update_and_fetch_dkey(struct io_test_args *arg, daos_epoch_t update_epoch,
 	iod.iod_recxs	= &rex;
 	iod.iod_nr	= 1;
 
+	/* Act */
 	rc = io_test_obj_update(arg, update_epoch, &dkey, &iod, &sgl, true);
 	if (rc)
 		goto exit;
 
+	/* Changes */
 	inc_cntr(arg->ta_flags);
 
 	memset(fetch_buf, 0, UPDATE_BUF_SIZE);
 	daos_iov_set(&val_iov, &fetch_buf[0], UPDATE_BUF_SIZE);
 
 	iod.iod_size = DAOS_REC_ANY;
-
-	if (!(arg->ta_flags & TF_REC_EXT) && arg->ta_flags & TF_USE_CSUM)
-		daos_csum_set(&csum, actual_csum_buf, UPDATE_CSUM_SIZE);
-
+	memset(actual_csum_buf, 0, sizeof(actual_csum_buf));
+	if (arg->ta_flags & TF_USE_CSUM)
+		daos_csum_set_multiple(&csum, actual_csum_buf,
```

Comment:
Is the plan to expose daos_csum_set_multiple? Are we seeing use cases for both csum_set and csum_set_multiple?
Why not just change csum_set to have all the parameters?